### PR TITLE
feat(route): add substack trending posts

### PR DESCRIPTION
**Pull request by bendobos**
## 该 PR 相关 Issue / Involved Issue

Close #

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/substack/trending/art
/substack/trending/art/true
/substack/trending/art/true/10
/substack/trending/staff-picks
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [x] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

### CODE REVIEW

File: lib/v2/substack/index.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+const SUBSTACK_API_BASE_URL = 'https://substack.com/api/v1';
+
+/**
+ * This route allows subscribing Substack "Explore" categories, available via https://substack.com/browse.
+ *
+ * Substack currently does not offer a public developer API, so the internal API utilized on the Substack website is used here.
+ * Since it is not an "official" API, keep in mind that the API calls might break at any time.
+ */
+module.exports = async (ctx) => {
+    const { category, allowPaid = 'false', limit = '36' } = ctx.params;
```
Comment (TonyRL) [inline lib/v2/substack/index.js:16]: You can use the common query parameter `limit` from `ctx.query.limit`
docs: https://docs.rsshub.app/en/parameter.html#limit-entries

File: lib/v2/substack/index.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    if (category === 'staff-picks') {
+        url = `${SUBSTACK_API_BASE_URL}/reader/posts/staff-picks`;
+    } else {
+        url = `${SUBSTACK_API_BASE_URL}/trending?category_id=${mappedCategory.id}&limit=${limit}`;
+    }
+
+    const { data } = await got(url, {
+        headers: {
+            accept: '*/*',
+            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/114.0',
```
Comment (TonyRL) [inline lib/v2/substack/index.js:40]: Every got request from RSSHub will use a randomised UA of Chrome on Mac. Are there any specific reasons to use the UA of Firefox on Windows?
https://github.com/DIYgod/RSSHub/blob/a8058f292167a03f3d064376d5741fcc0c671ff1/lib/config.js#L49

File: lib/v2/substack/templates/description.art [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{{link}}
+
+{{subtitle}}
+{{if subtitle}}
+    <h2>{{subtitle}}</h2>
```
Comment (TonyRL) [inline lib/v2/substack/templates/description.art:5]: `subtitle` will be duplicated.

File: lib/v2/substack/templates/description.art [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        {{reactions}} ❤
+        {{if tags.length > 0}}
+            | tags:
+            {{each tags}}
+                {{$value.name}}
+                {{if $index != tags.length - 1}},{{/if}}
+            {{/each}}
+        {{/if}}
+    </small>
+</p>
```
Comment (TonyRL) [inline lib/v2/substack/templates/description.art:30]: If you want a line break, you need to use `<br>` explicitly.

File: lib/v2/substack/templates/description.art [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{{link}}
```
Comment (TonyRL) [inline lib/v2/substack/templates/description.art:1]: Don't put the `link` in the content. It should appear in `ctx.state.data.item.link` only.

File: lib/v2/substack/index.js (hunk 2 above)
Comment (bendobos) [reply lib/v2/substack/index.js:40]: Ah, I wasn't aware of that. 
I just chose Firefox on Windows to put any user agent here. 

I will remove that.

**Pull request closed without merging**